File: src/main/java/dev/lyzev/api/jna/DwmApi.java
/*
 * Copyright (c) 2024. Schizoid
 * All rights reserved.
 */

package dev.lyzev.api.jna;

import com.sun.jna.Library;
import com.sun.jna.Native;
import com.sun.jna.PointerType;
import com.sun.jna.platform.win32.WinDef;

public interface DwmApi extends Library {

    public static final DwmApi INSTANCE = Native.loadLibrary("dwmapi", DwmApi.class);

    int DwmSetWindowAttribute(WinDef.HWND hwnd, int dwAttribute, PointerType pvAttribute, int cbAttribute);

}

File: src/main/kotlin/dev/lyzev/api/jna/DwmApi.kt
/*
 * Copyright (c) 2024. Schizoid
 * All rights reserved.
 */
package dev.lyzev.api.jna

import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.PointerType
import com.sun.jna.platform.win32.WinDef.HWND

interface DwmApi : Library {

    fun DwmSetWindowAttribute(hwnd: HWND, dwAttribute: Int, pvAttribute: PointerType, cbAttribute: Int): Int

    companion object: DwmApi by Native.load("dwmapi", DwmApi::class.java)
}
